src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -470,7 +470,7 @@ private static SyntaxNode CreateImport(SyntaxGenerator syntaxGenerator, string n
                 .ConfigureAwait(false);
             var solutionWithChangedNamespace = documentWithNewNamespace.Project.Solution;
 
-            var refLocationGroups = refLocationsInOtherDocuments.GroupBy(loc => loc.Document.Id);
+            var refLocationGroups = refLocationsInOtherDocuments.Where(loc => solutionWithChangedNamespace.ContainsDocument(loc.Document.Id)).GroupBy(loc => loc.Document.Id);
```

Comment:
So either FindReferences returns a reference in a Document that's not a part of the solution (seems implausible, but would be bad if true), or somehow `FixDeclarationDocumentAsync` caused the referencing document to be removed (a quick glance suggests this is not the case).

Could we add some NFW to collect more information?

Reply:
Added